pkgs/development/interpreters/elixir/generic-builder.nix

```diff
@@ -0,0 +1,77 @@
+{ pkgs, stdenv, fetchFromGitHub, erlang, rebar, makeWrapper,
+  coreutils, curl, bash, debugInfo ? false }:
+
+{ baseName ? "elixir"
+, version
+, minimumOTPVersion
+, sha256 ? null
+, rev ? "v${version}"
+, src ? fetchFromGitHub { inherit rev sha256; owner = "elixir-lang"; repo = "elixir"; }
+}:
+
+let
+  inherit (stdenv.lib) getVersion versionAtLeast;
+
+in
+  assert versionAtLeast (getVersion erlang) minimumOTPVersion;
```

Comment:
I just re-ran nox-review overnight.

```
/nix/store/gmqllgyna6kav9zz770fkczlh928hd8f-erlang-20.0-odbc
/nix/store/5421cdrj4zf5dkqfld4pcba2jfjrk8k4-erlang-19.3-javac-odbc
/nix/store/hs7jj5niyx13awgi73kqdsb00a62jz8f-erlang-18.3.4.4-javac
/nix/store/gfks6rlfrg37k20gmw3by6j5pv32lwhi-erlang-16B03-1
/nix/store/b39vr4vay0cx9mvlzwb9a28yp3lc9gs6-erlang-17.5-javac-odbc
/nix/store/55alxr80yb0s0k4cm7a3svw7kh4rwz42-elixir-1.5.0-rc.0
/nix/store/dgi75xy4xngpjvn9ijzkabpp1ppwf5w6-erlang-17.5-odbc
/nix/store/8d2m82zrvq8r247zhx99yj63vmrb8zaw-erlang-19.3-javac
/nix/store/pmiwscdddsxyvc2la6fiygjh9nqgnr6q-erlang-20.0-javac-odbc
/nix/store/sjz5ijsvrysas3swr1aygay85sy6ayb4-erlang-17.5-javac
/nix/store/29gn1vapx5mqdka3cixh8flzgnspggwr-erlang-18.3.4.4-odbc
/nix/store/5b6djrg39bbrfd8k9dfpcsa8vrkygjpd-erlang-16B03-1-odbc
/nix/store/ddc3ix61gs6nml93wr1yddfawlyxpg0f-erlang-20.0-javac
/nix/store/b2c6whz71kxaa9jrg7zc25l8s1a1zcr6-elixir-1.3.4
/nix/store/dy1g580h1cy06gzh2pv2z99mf0c3kw4j-erlang-16B02-odbc
/nix/store/nmi33wz63dwld6xnxn7w70cpsbiljs9f-erlang-18.3.4.4-javac-odbc
/nix/store/asrc5783fmkjrvs1k6r2fbh1p5m6b51x-erlang-19.3-odbc
Result in /var/folders/lp/n8j6lzmj5dj34zgzb4zxw35c0000gn/T/nox-review-cooxkh3s
total 68
lrwxr-xr-x 1 woodjk staff 60 Jun 27 00:36 result -> /nix/store/gmqllgyna6kav9zz770fkczlh928hd8f-erlang-20.0-odbc
lrwxr-xr-x 1 woodjk staff 61 Jun 27 00:36 result-10 -> /nix/store/sjz5ijsvrysas3swr1aygay85sy6ayb4-erlang-17.5-javac
lrwxr-xr-x 1 woodjk staff 64 Jun 27 00:36 result-11 -> /nix/store/29gn1vapx5mqdka3cixh8flzgnspggwr-erlang-18.3.4.4-odbc
lrwxr-xr-x 1 woodjk staff 63 Jun 27 00:36 result-12 -> /nix/store/5b6djrg39bbrfd8k9dfpcsa8vrkygjpd-erlang-16B03-1-odbc
lrwxr-xr-x 1 woodjk staff 61 Jun 27 00:36 result-13 -> /nix/store/ddc3ix61gs6nml93wr1yddfawlyxpg0f-erlang-20.0-javac
lrwxr-xr-x 1 woodjk staff 56 Jun 27 00:36 result-14 -> /nix/store/b2c6whz71kxaa9jrg7zc25l8s1a1zcr6-elixir-1.3.4
lrwxr-xr-x 1 woodjk staff 61 Jun 27 00:36 result-15 -> /nix/store/dy1g580h1cy06gzh2pv2z99mf0c3kw4j-erlang-16B02-odbc
lrwxr-xr-x 1 woodjk staff 70 Jun 27 00:36 result-16 -> /nix/store/nmi33wz63dwld6xnxn7w70cpsbiljs9f-erlang-18.3.4.4-javac-odbc
lrwxr-xr-x 1 woodjk staff 60 Jun 27 00:36 result-17 -> /nix/store/asrc5783fmkjrvs1k6r2fbh1p5m6b51x-erlang-19.3-odbc
lrwxr-xr-x 1 woodjk staff 66 Jun 27 00:36 result-2 -> /nix/store/5421cdrj4zf5dkqfld4pcba2jfjrk8k4-erlang-19.3-javac-odbc
lrwxr-xr-x 1 woodjk staff 65 Jun 27 00:36 result-3 -> /nix/store/hs7jj5niyx13awgi73kqdsb00a62jz8f-erlang-18.3.4.4-javac
lrwxr-xr-x 1 woodjk staff 58 Jun 27 00:36 result-4 -> /nix/store/gfks6rlfrg37k20gmw3by6j5pv32lwhi-erlang-16B03-1
lrwxr-xr-x 1 woodjk staff 66 Jun 27 00:36 result-5 -> /nix/store/b39vr4vay0cx9mvlzwb9a28yp3lc9gs6-erlang-17.5-javac-odbc
lrwxr-xr-x 1 woodjk staff 61 Jun 27 00:36 result-6 -> /nix/store/55alxr80yb0s0k4cm7a3svw7kh4rwz42-elixir-1.5.0-rc.0
lrwxr-xr-x 1 woodjk staff 60 Jun 27 00:36 result-7 -> /nix/store/dgi75xy4xngpjvn9ijzkabpp1ppwf5w6-erlang-17.5-odbc
lrwxr-xr-x 1 woodjk staff 61 Jun 27 00:36 result-8 -> /nix/store/8d2m82zrvq8r247zhx99yj63vmrb8zaw-erlang-19.3-javac
lrwxr-xr-x 1 woodjk staff 66 Jun 27 00:36 result-9 -> /nix/store/pmiwscdddsxyvc2la6fiygjh9nqgnr6q-erlang-20.0-javac-odbc
```

I didn't use the Erlang specific `getVersion` for two reasons.

1. Elixir only builds on Erlang 18+. And as you point out, you wrote that function because of R16.
2. I don't think it would have caused any issues, but using that version would have introduced a circular dependency. I'm not sure if this sort of thing is frowned upon in the Nix world.

I don't mind using the Erlang specific one if my second point is not considered a problem, or if we move the function elsewhere. I just figured it would be nice to use the standard set of functions where possible so that anyone looking at the code in the future would not have to lookup the definition of another function.